File: DotNet/core_monitoring_tests/Persistence/TestMethodCallPO.cs
using System.Reflection;
using NUnit.Framework;

namespace Org.NMonitoring.Core.Persistence.Tests
{
    [TestFixture]
    public class TestMethodCallPO
    {

        [Test]
        public void testUpdateChildrenWhenCreateWithParent()
        {
            MethodCallPO tParent = new MethodCallPO(null, this.GetType().FullName, "builNewFullFlow", "GrDefault",
                            new ParameterInfo[0]);

            MethodCallPO tChild = new MethodCallPO(tParent, this.GetType().FullName, "builNewFullFlow2",
                            "GrChild1", new ParameterInfo[0]);

            Assert.IsNull(tParent.Parent);
            Assert.AreEqual(1, tParent.Children.Count);
            Assert.AreSame(tParent, tChild.Parent);

            tParent.RemoveChildren(tChild);
            Assert.IsNull(tParent.Parent);
            Assert.AreEqual(0, tParent.Children.Count);
            Assert.IsNull(tChild.Parent);
        }

        [Test]
        public void testUpdateChildrenWhenAddParent()
        {
           MethodCallPO tParent = new MethodCallPO(null, this.GetType().FullName, "builNewFullFlow", "GrDefault",
                            new ParameterInfo[0]);

            MethodCallPO tChild = new MethodCallPO(null, this.GetType().FullName, "builNewFullFlow2",
                            "GrChild1", new ParameterInfo[0]);

            Assert.IsNull(tParent.Parent);
            Assert.AreEqual(0, tParent.Children.Count);
            Assert.IsNull(tChild.Parent);

            tChild.Parent=tParent;
            Assert.IsNull(tParent.Parent);
            Assert.AreEqual(1, tParent.Children.Count);
            Assert.AreSame(tParent, tChild.Parent);

            tChild.Parent=null;
            tChild.Parent=null; // On teste avec null 2 fois...
            Assert.IsNull(tParent.Parent);
            Assert.AreEqual(0, tParent.Children.Count);
            Assert.IsNull(tChild.Parent);

            tChild.Parent=tParent;
            Assert.IsNull(tParent.Parent);
            Assert.AreEqual(1, tParent.Children.Count);
            Assert.AreSame(tParent, tChild.Parent);
        }
    }
}
 

File: framework/core_monitoring/src-unit-test/org/jmonitoring/core/persistence/TestMethodCallPO.java
package org.jmonitoring.core.persistence;

import org.jmonitoring.core.dao.PersistanceTestCase;
import org.jmonitoring.core.dao.TestExecutionFlowDAO;

/***************************************************************************
 * Copyright 2005 Philippe Kernevez All rights reserved.                   *
 * Please look at license.txt for more license detail.                     *
 **************************************************************************/

public class TestMethodCallPO extends PersistanceTestCase
{
    public void testUpdateChildrenWhenCreateWithParent()
    {
        MethodCallPO tParent = new MethodCallPO(null, TestExecutionFlowDAO.class.getName(), "builNewFullFlow", "GrDefault",
                        new Object[0]);

        MethodCallPO tChild = new MethodCallPO(tParent, TestExecutionFlowDAO.class.getName(), "builNewFullFlow2",
                        "GrChild1", new Object[0]);    
        assertNull(tParent.getParentMethodCall());
        assertEquals(1, tParent.getChildren().size());
        assertSame(tParent, tChild.getParentMethodCall());
        
        tParent.removeChildren(tChild);
        assertNull(tParent.getParentMethodCall());
        assertEquals(0, tParent.getChildren().size());
        assertNull(tChild.getParentMethodCall());
        
        
    }
    
    public void testUpdateChildrenWhenAddParent()
    {
        MethodCallPO tParent = new MethodCallPO(null, TestExecutionFlowDAO.class.getName(), "builNewFullFlow", "GrDefault",
                        new Object[0]);

        MethodCallPO tChild = new MethodCallPO(null, TestExecutionFlowDAO.class.getName(), "builNewFullFlow2",
                        "GrChild1", new Object[0]);    

        assertNull(tParent.getParentMethodCall());
        assertEquals(0, tParent.getChildren().size());
        assertNull(tChild.getParentMethodCall());
        
        tChild.setParentMethodCall(tParent);
        assertNull(tParent.getParentMethodCall());
        assertEquals(1, tParent.getChildren().size());
        assertSame(tParent, tChild.getParentMethodCall());
        
        tChild.setParentMethodCall(null);
        tChild.setParentMethodCall(null); // On teste avec null 2 fois...
        assertNull(tParent.getParentMethodCall());
        assertEquals(0, tParent.getChildren().size());
        assertNull(tChild.getParentMethodCall());
        
        tChild.setParentMethodCall(tParent);
        assertNull(tParent.getParentMethodCall());
        assertEquals(1, tParent.getChildren().size());
        assertSame(tParent, tChild.getParentMethodCall());
        

    }
}
